libsplinter/src/biome/rest_api/actix/key_management.rs

```diff
@@ -35,31 +37,62 @@ pub fn make_key_management_route(
     key_store: Arc<dyn KeyStore>,
     secret_manager: Arc<dyn SecretManager>,
 ) -> Resource {
-    Resource::build("/biome/keys")
-        .add_request_guard(ProtocolVersionRangeGuard::new(
+    let resource =
+        Resource::build("/biome/keys").add_request_guard(ProtocolVersionRangeGuard::new(
             protocol::BIOME_KEYS_PROTOCOL_MIN,
             protocol::BIOME_PROTOCOL_VERSION,
-        ))
-        .add_method(
-            Method::Post,
-            handle_post(
-                rest_config.clone(),
-                key_store.clone(),
-                secret_manager.clone(),
-            ),
-        )
-        .add_method(
-            Method::Get,
-            handle_get(
-                rest_config.clone(),
-                key_store.clone(),
-                secret_manager.clone(),
-            ),
-        )
-        .add_method(
-            Method::Patch,
-            handle_patch(rest_config, key_store, secret_manager),
-        )
+        ));
+    #[cfg(feature = "authorization")]
+    {
+        resource
+            .add_method(
+                Method::Post,
+                NO_CHECK_PERMISSION.into(),
```

Comment:
I wonder if we should define

enum Permission {
    Allow(String),
    AllowAll,
}

That would make this look like:

.add_method(Method::Post, Permission::AllowAll, ...).

other places we would use

const CIRCUIT_READ_PERMISSION: Permission = Permission::Allow("circuit.read");

This would also type the permission arguments in functions.

Reply:
Updated permissions to be an enum similar to the one you suggested.